<?xml version="1.0"?>
<launch>

    <node pkg="sui-201-ros-pkg" type="powerPub.py" name="power_publisher" />

	<node pkg="sbus_serial" type="sbus_serial_node" name="sbus_node" output="screen">
		<param name="port" value="/dev/sbus" />
		<param name="refresh_rate_hz" value="25" />
		<param name="rxMinValue" value="695" />
		<param name="rxMaxValue" value="1509" />
		<param name="outMinValue" value="0" />
		<param name="outMaxValue" value="1000" />
		<param name="silentOnFailsafe" value="false" />
		<param name="enableChannelNum" value="-1" />
		<param name="enableChannelProportionalMin" type="double" value="0.0" />
		<param name="enableChannelProportionalMax" type="double" value="0.0" />

	</node>

    <!-- spawn controller node -->
    <node pkg="fwd_4ws_lower_control" type="fwd_control" name="fwd_control" output="screen">
        <param name="speedMax"     type="double"    value="2.5"/>
        <param name="angularSpeedMax"     type="double"    value="6.0"/>
		<param name="minSpeedThreshold"     type="double"    value="0.05"/>
    </node>

    <!-- PlotJuggle -->
    <!-- <node name="plotjuggler_with_layout" 
          pkg="plotjuggler"
          type="plotjuggler"
          args="-remove-layout $(find pro_lower_controller)/rviz/plotjuggler.xml"/> -->

	<!-- broadcast base_link -> laser tf -->
    <node pkg="tf2_ros" type="static_transform_publisher" 
        name="laser_static_transform_publisher" args="0.15 0 0.15 3.14159 0  0  base_footprint laser" />     

	<!-- broadcast camera_link -> base_link tf -->
    <node pkg="tf2_ros" type="static_transform_publisher" 
          name="base_static_transform_publisher" 
          args="0.2 0 0.15 0  0 0 base_footprint camera_pose_frame" /> 
</launch>
